util/src/main/scala/scala/scalanative/io/ByteBufferPool.scala

```diff
@@ -7,7 +7,7 @@ final class ByteBufferPool {
   private var buffers: List[ByteBuffer] = Nil
 
   private def alloc(): ByteBuffer = {
-    ByteBuffer.allocateDirect(64 * 1024 * 1024)
+    ByteBuffer.allocateDirect(8 * 1024 * 1024)
```

Comment:
Solution proposed here is a bit of bandaid, we'll probably rework this more fundamentally later on. There is quite a bit of cruft that has been accumulated around here and quite a few things that can be improved, but that would require drastic changes which are out of scope for this pr.